### PR TITLE
chore(makefile): enhance reproducible build

### PR DESCRIPTION
**Pull request by developer-guy**
Signed-off-by: Batuhan Apaydın <batuhan.apaydin@trendyol.com>
Co-authored-by: Morten Linderud <morten@linderud.pw>

**1. Issue, if available:**

**2. Description of changes:**

Based on [reproducible builds](https://reproducible-builds.org/docs/source-date-epoch/) documents, we need to consider using SOURCE_DATE_EPOCH which is set by downstream build systems while preparing the BUILD_DATE. So, this PR will add that support.

**3. How was this change tested?**

**4. Does this change impact docs?**
- [ ] Yes, PR includes docs updates
- [ ] Yes, issue opened: *link to issue*
- [x] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

cc: @Foxboron

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -2,8 +2,18 @@ RELEASE_REPO ?= public.ecr.aws/karpenter
...
 
+# https://reproducible-builds.org/docs/source-date-epoch/
+DATE_FMT = %Y-%m-%dT%H:%M:%SZ
+ifdef SOURCE_DATE_EPOCH
+    BUILD_DATE ?= $(shell date -u -d "@$(SOURCE_DATE_EPOCH)" "$(DATE_FMT)" 2>/dev/null || date -u -r "$(SOURCE_DATE_EPOCH)" "$(DATE_FMT)" 2>/dev/null || date -u "$(DATE_FMT)")
+else
+    BUILD_DATE ?= $(shell date "$(DATE_FMT)")
+endif
+set colorscheme
+set colorscheme
```
Comment (Foxboron) [inline Makefile:13]: Typo I believe :)?

File: Makefile [hunk 2]
```diff
@@ -2,8 +2,18 @@ RELEASE_REPO ?= public.ecr.aws/karpenter
 RELEASE_VERSION ?= $(shell git describe --tags --always)
 RELEASE_PLATFORM ?= --platform=linux/amd64,linux/arm64
 
+# https://reproducible-builds.org/docs/source-date-epoch/
+DATE_FMT = %Y-%m-%dT%H:%M:%SZ
```
Comment (Foxboron) [inline Makefile:6]:
```suggestion
DATE_FMT = +%Y-%m-%dT%H:%M:%SZ
```

Comment (developer-guy) [reply Makefile:6]: great catch, I used vim to do updates, sometimes I catch those things 🙋🏻‍♂️ Thanks for having my back 🤝

**Review by ellistarn — APPROVED** [review]

**Comment by ellistarn** [comment]
Thank you!

**Pull request merged**